component/main.jsonnet

```
@@ -30,7 +30,44 @@ local opaqueSecrets() = [
   for name in std.objectFields(params.opaque)
 ];
 
+local tlsSecrets() = [
+  local values = std.get(params.tls, name);
+  {
+    apiVersion: 'v1',
+    kind: 'Secret',
+    metadata: {
+      name: namespacedName(name, '').name,
+      namespace: namespacedName(name, '').namespace,
+    },
+    type: 'kubernetes.io/tls',
+    data: {
+      'tls.crt': values.key,
+      'tls.key': values.cert,
+    },
+  }
+  for name in std.objectFields(params.tls)
+];
+
+local dockercfgSecrets() = [
+  local values = std.get(params.dockercfg, name);
+  {
+    apiVersion: 'v1',
+    kind: 'Secret',
+    metadata: {
+      name: namespacedName(name, '').name,
+      namespace: namespacedName(name, '').namespace,
+    },
+    type: 'kubernetes.io/dockercfg',
+    data: {
```

This should be `stringData` to support Vault references.

```suggestion
    stringData: {
```

---

fixed